src/python/grpcio_tests/tests/interop/server_ssl_cert_reload_test/_test.py

```diff
@@ -0,0 +1,411 @@
+    return
+
+def assert_client_failure(stdout, stderr, client_rejects_server):
+    # "client_rejects_server": true if the failure is because client
+    # rejects server, false if because server rejects client
+    # print 'checking stdout...'
+    assert stdout == '', 'unexpected stdout: {}'.format(stdout)
+    if client_rejects_server:
+        # print 'checking for "handshake failed" msg...'
+        # if server rejects client, then client won't see this msg
+        assert 'Handshake failed with fatal error' in stderr
+    assert not 'Greeter client received' in stderr
+    return
+
+def check_background_client_success(stdout, stderr,
+                                    before_switch_timestamp, after_switch_timestamp):
+    assert stderr == '', stderr
+    lines = stdout.splitlines()
+    assert len(lines) == 2
+    assert lines[0].startswith('Greeter client received: Hello, '), lines[0]
+    assert int(lines[0].split()[4]) < before_switch_timestamp
+    assert lines[1].startswith('Greeter client received: Hello, '), lines[1]
+    assert int(lines[1].split()[4]) > after_switch_timestamp
+
+
+
+class ServerSSLCertReloadTest(unittest.TestCase):
+
+    def test_server_ssl_cert_reload_basic(self):
+        switch_cert_on_client_num = 9
+
+        # use tempfiles to contain the client's cert plus intermediate
+        # cert from the two cert hierarchies
+
+        client_key_1_fpath = _get_abs_path('cert_hier_1/intermediate/private/client.key.pem')
+        client_chain_1_fp = tempfile.NamedTemporaryFile(suffix='.pem')
+        client_chain_1_fp.write('''{}
+{}
+'''.format(open(_get_abs_path('cert_hier_1/intermediate/certs/client.cert.pem')).read(),
+           open(_get_abs_path('cert_hier_1/intermediate/certs/intermediate.cert.pem')).read(),
+           ))
+        client_chain_1_fp.flush()
+
+        client_key_2_fpath = _get_abs_path('cert_hier_2/intermediate/private/client.key.pem')
+        client_chain_2_fp = tempfile.NamedTemporaryFile(suffix='.pem')
+        client_chain_2_fp.write('''{}
+{}
+'''.format(open(_get_abs_path('cert_hier_2/intermediate/certs/client.cert.pem')).read(),
+           open(_get_abs_path('cert_hier_2/intermediate/certs/intermediate.cert.pem')).read(),
+           ))
+        client_chain_2_fp.flush()
+
+        server_ca_1_fpath = _get_abs_path('cert_hier_1/certs/ca.cert.pem')
+        server_ca_2_fpath = _get_abs_path('cert_hier_2/certs/ca.cert.pem')
+
+        greeter_client_path = _get_abs_path('greeter_client.py')
+
+        require_client_auth = True
+
+        devnull_fp = open(os.devnull)
+        server_process = subprocess.Popen(
+            ['python', _get_abs_path('greeter_server.py'),
+             str(switch_cert_on_client_num), str(require_client_auth)],
+            stdout=devnull_fp, stderr=devnull_fp)
+        processes.append(server_process)
+        # give server a second to initialize
+        import time
+        time.sleep(1)
+
+        print 'things should work...',
```

Comment:
Please don't include print statements in unit tests?

Reply:
done